org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalDescriptionProvider.java

```diff
@@ -319,20 +328,33 @@ private static final StringBuilder appendParameterSignature(StringBuilder buffer
 	 */
 	private void createMethodProposalLabel(CompletionProposal methodProposal, CompletionItem item) {
 		StringBuilder description = this.createMethodProposalDescription(methodProposal);
+		String proposalName = String.valueOf(methodProposal.getName());
+		boolean skipDetail = false;
```

Comment:
Did you do this because you noticed that the regular `detail` field was being set for completion items that were using the `labelDetails` field, so you tried to preserve that behaviour for regular methods ? But opted to eliminate it for collapsed types ?

According to https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem I don't really see anything that says label details make the regular `detail` field obsolete so I guess we opted to keep it around.

Reply:
Yes, I didn't want to alter the behaviour for non-collapsed proposals, but removed it for the collapsed proposals because it contained proposal details that would only be related to one of multiple collapsed proposals. There was nothing relevant to include in the field so I just left it as null.

Reply:
Yeah, I guess that's fine. Also `skipDetail` seems to be `true` only when a type should be collapsed & is not a constructor so you probably could have used the exact same check.

I guess I was aiming towards moving the `if (!skipDetail)` block into the block above, but because it's used in 2/4 cases, might as well save space like we are now.